### PR TITLE
orcania: add orcania/2.3.1 recipe

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **orcania/2.3.1**

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/orcania/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        "base64url": True,
+    }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
```
Comment (jwillikers) [inline recipes/orcania/all/conanfile.py:36]:
```suggestion
            try:
                del self.options.fPIC
            except Exception:
                pass
```

File: recipes/orcania/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        # TODO: to remove in conan v2 once cmake_find_package* generators removed
+        self._create_cmake_module_alias_targets(
+            os.path.join(self.package_folder, self._module_file_rel_path),
+            {} if self.options.shared else {"Orcania::Orcania-static": "Orcania::Orcania"}
+        )
+
+    def _create_cmake_module_alias_targets(self, module_file, targets):
+        content = ""
+        for alias, aliased in targets.items():
+            content += textwrap.dedent("""\
```
Comment (jwillikers) [inline recipes/orcania/all/conanfile.py:102]: Python F-string here?

File: recipes/orcania/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+                if(TARGET {aliased} AND NOT TARGET {alias})
+                    add_library({alias} INTERFACE IMPORTED)
+                    set_property(TARGET {alias} PROPERTY INTERFACE_LINK_LIBRARIES {aliased})
+                endif()
+            """.format(alias=alias, aliased=aliased))
+        save(self, module_file, content)
+
+    @property
+    def _module_file_rel_path(self):
+        return os.path.join("lib", "cmake", "conan-official-{}-targets.cmake".format(self.name))
```
Comment (jwillikers) [inline recipes/orcania/all/conanfile.py:112]: A Python F-string would work well here.

**Review by jwillikers — CHANGES_REQUESTED** [review]

**Review by uilianries — APPROVED** [review]

File: recipes/orcania/all/conanfile.py (hunk 3 above)
Comment (madebr) [reply recipes/orcania/all/conanfile.py]: Addressed in https://github.com/conan-io/conan-center-index/pull/13589/commits/ca1863cb824f2b86cae211f60e9fc8da96aa7c0d

File: recipes/orcania/all/conanfile.py (hunk 1 above)
Comment (madebr) [reply recipes/orcania/all/conanfile.py]: Addressed in https://github.com/conan-io/conan-center-index/pull/13589/commits/ca1863cb824f2b86cae211f60e9fc8da96aa7c0d

File: recipes/orcania/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            try:
+                del self.settings.compiler.libcxx
```
Comment (madebr) [inline recipes/orcania/all/conanfile.py:37]:
```suggestion
                del self.options.fPIC
```

**Review by jwillikers — DISMISSED** [review]

File: recipes/orcania/all/conandata.yml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+sources:
+  "2.3.1":
+    url: "https://github.com/babelouest/orcania/archive/refs/tags/v2.3.1.tar.gz"
+    sha256: "bbf08d563528b8ab88dd4b0e67aeb4e7c4fc9f19dcd1a0346b773cf492f7612b"
+patches:
+  "2.3.1":
+    - patch_file: "patches/2.3.1-0001-shared-static-conan.patch"
+      patch_description: "Build shared and static libraries + fix MSVC support"
+      patch_type: "conan"
+    - patch_file: "patches/2.3.1-0002-no-m.patch"
```
Comment (prince-chrismc) [inline recipes/orcania/all/conandata.yml:10]: please submit this upstream 🙏

File: recipes/orcania/all/conandata.yml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+sources:
+  "2.3.1":
+    url: "https://github.com/babelouest/orcania/archive/refs/tags/v2.3.1.tar.gz"
+    sha256: "bbf08d563528b8ab88dd4b0e67aeb4e7c4fc9f19dcd1a0346b773cf492f7612b"
+patches:
+  "2.3.1":
+    - patch_file: "patches/2.3.1-0001-shared-static-conan.patch"
```
Comment (prince-chrismc) [inline recipes/orcania/all/conandata.yml:7]: this should also be upstreamed 🙏 ❤️

File: recipes/orcania/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+    description = "Potluck with different functions for different purposes that can be shared among C programs"
+    homepage = "https://github.com/babelouest/orcania"
+    topics = ("utility", "functions", )
+    license = "LGPL-2.1-or-later"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "base64url": [True, False],
```
Comment (prince-chrismc) [inline recipes/orcania/all/conanfile.py:22]: maybe enable or use https://github.com/conan-io/conan-center-index/blob/master/docs/packaging_policy.md#recommended-feature-options-names?

File: recipes/orcania/all/conandata.yml (hunk 5 above)
Comment (madebr) [reply recipes/orcania/all/conandata.yml:10]: Way ahead of you  https://github.com/babelouest/orcania/pull/30 :wink:

File: recipes/orcania/all/conanfile.py (hunk 7 above)
Comment (madebr) [inline recipes/orcania/all/conanfile.py:22]:
```suggestion
        "enable_base64url": [True, False],
```

File: recipes/orcania/all/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["BUILD_BASE64URL"] = self.options.base64url
```
Comment (madebr) [inline recipes/orcania/all/conanfile.py:65]:
```suggestion
        tc.variables["BUILD_BASE64URL"] = self.options.enable_base64url
```

File: recipes/orcania/all/conanfile.py [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "base64url": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "base64url": True,
```
Comment (madebr) [inline recipes/orcania/all/conanfile.py:27]:
```suggestion
        "enable_base64url": True,
```

File: recipes/orcania/all/conanfile.py [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+            del self.settings.compiler.libcxx
+        except Exception:
+            pass
+        try:
+            del self.settings.compiler.cppstd
+        except Exception:
+            pass
+
+    def requirements(self):
+        if is_msvc(self) and self.options.base64url:
```
Comment (madebr) [inline recipes/orcania/all/conanfile.py:50]:
```suggestion
        if is_msvc(self) and self.options.enable_base64url:
```

File: recipes/orcania/all/conandata.yml (hunk 6 above)
Comment (madebr) [reply recipes/orcania/all/conandata.yml:7]: https://github.com/babelouest/orcania/issues/31

**Review by jwillikers — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by jwillikers — DISMISSED** [review]

**Review by jwillikers — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Comment by prince-chrismc** [comment]
yummy new package 🤤

**Pull request merged**